src/frontends/tensorflow/src/translate_session.cpp

```diff
@@ -570,6 +602,10 @@ void TranslateSession::translate_graph(const ov::frontend::InputModel::Ptr& inpu
                                             operation_name + "node specified as custom output does not exist");
                 auto result_node = std::make_shared<ov::opset8::Result>(node_outputs[port_index]);
                 result_node->set_friendly_name(model_output_name);
+                if (set_names_with_indices) {
+                    result_node->get_output_tensor(0).set_names(
+                        {make_output_tensor_name(result_node, operation_name, port_index)});
+                }
```

Comment:
I am not sure that we really need `make_output_tensor_name`. I think we should modify `adjust_saved_model_names` and `set_node_name`.

Reply:
Moved alignment to `adjust_saved_model_names()`. I kept `make_output_tensor_name()`, as it contains the logic of making a resulting name with ":0". Please check.